<rpcsvc/nis.h> is just the bare minimum required to compile, along with
the original license header.

--- origsrc/libtirpc-0.3.2/Makefile.am	2015-06-16 10:35:08.000000000 -0500
+++ src/libtirpc-0.3.2/Makefile.am	2015-10-13 13:51:45.088825300 -0500
@@ -4,6 +4,7 @@ ACLOCAL_AMFLAGS = -I m4
 noinst_HEADERS	       = tirpc/reentrant.h \
 			 tirpc/getpeereid.h \
 			 tirpc/libc_private.h \
+			 tirpc/rpcsvc/nis.h \
 			 tirpc/un-namespace.h
 
 nobase_include_HEADERS = tirpc/netconfig.h \
--- origsrc/libtirpc-0.3.2/src/getpublickey.c	2015-06-16 10:35:08.000000000 -0500
+++ src/libtirpc-0.3.2/src/getpublickey.c	2015-10-13 13:43:54.996631200 -0500
@@ -38,8 +38,10 @@
 #include <pwd.h>
 #include <rpc/rpc.h>
 #include <rpc/key_prot.h>
+#ifdef YP
 #include <rpcsvc/yp_prot.h>
 #include <rpcsvc/ypclnt.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 
--- origsrc/libtirpc-0.3.2/tirpc/rpcsvc/nis.h	1969-12-31 18:00:00.000000000 -0600
+++ src/libtirpc-0.3.2/tirpc/rpcsvc/nis.h	2015-10-13 13:43:14.978049400 -0500
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2010, Oracle America, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials
+ *       provided with the distribution.
+ *     * Neither the name of the "Oracle America, Inc." nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *   FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *   COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *   INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ *   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ *   GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *   INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *   WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _RPCSVC_NIS_H
+#define _RPCSVC_NIS_H 1
+
+#include <rpc/rpc.h>
+
+#define NIS_PK_NONE 0
+#define NIS_PK_DH 1
+#define NIS_PK_RSA 2
+#define NIS_PK_KERB 3
+#define NIS_PK_DHEXT 4
+
+typedef char *nis_name;
+
+struct endpoint {
+    char *uaddr;
+    char *family;
+    char *proto;
+};
+typedef struct endpoint endpoint;
+
+struct nis_server {
+    nis_name name;
+    struct {
+        u_int ep_len;
+        endpoint *ep_val;
+    } ep;
+    uint32_t key_type;
+    netobj pkey;
+};
+typedef struct nis_server nis_server;
+
+#endif /* ! _RPCSVC_NIS_H */
